### ISSUE TITLE
count+order by not support

### ISSUE THREAD

**Issue by dbzzzzz** [apache/shardingsphere#10563]
### Which version of ShardingSphere did you use?
shardingsphere 5
### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy
### Expected behavior
select count(1) from t_order order by order_id desc;
should return correct result.
### Actual behavior
ERROR 1140 (42000): In aggregated query without GROUP BY, expression #2 of SELECT list contains nonaggregated column 'testdb_0.t_order_0.order_id'; this is incompatible with sql_mode=only_full_group_by

**Comment by strongduanmu** [comment]
@dbzzzzz It can be seen from the error information that this error is thrown when PG executes SQL, and `SS` is responsible for routing, rewriting, executing and merging results.

actual sql:

```
[INFO ] 2021-05-31 17:49:45.008 [Connection-2-ThreadExecutor] ShardingSphere-SQL - Logic SQL: select count(1) from t_order order by order_id desc;
[INFO ] 2021-05-31 17:49:45.011 [Connection-2-ThreadExecutor] ShardingSphere-SQL - SQLStatement: PostgreSQLSelectStatement(limit=Optional.empty, lock=Optional.empty)
[INFO ] 2021-05-31 17:49:45.011 [Connection-2-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_0 ::: select count(1) , order_id AS ORDER_BY_DERIVED_0 from t_order_0 order by order_id desc;
[INFO ] 2021-05-31 17:49:45.011 [Connection-2-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_0 ::: select count(1) , order_id AS ORDER_BY_DERIVED_0 from t_order_1 order by order_id desc;
[INFO ] 2021-05-31 17:49:45.011 [Connection-2-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_1 ::: select count(1) , order_id AS ORDER_BY_DERIVED_0 from t_order_0 order by order_id desc;
[INFO ] 2021-05-31 17:49:45.011 [Connection-2-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_1 ::: select count(1) , order_id AS ORDER_BY_DERIVED_0 from t_order_1 order by order_id desc;
```
error message:
```
org.postgresql.util.PSQLException: ERROR: column "t_order_0.order_id" must appear in the GROUP BY clause or be used in an aggregate function
  位置：19
```

**Comment by strongduanmu** [comment]
After we change the sql to `select count(1) from t_order group by order_id order by order_id desc;`, another error will occur:

```
ERROR:  Can't find index: AggregationProjection(type=COUNT, innerExpression=(1), alias=Optional.empty, derivedAggregationProjections=[], index=-1), please add alias for aggregate selections
```
This exception is because the `columnLabel` that in the `columnLabelIndexMap` returned by PG is `count`, and the `columnLabel` processed by SS is `count(1)`, the inconsistency of the `columnLabel` results in an exception in the verification. Considering the different processing way of MySQL, we need to distinguish between different database dialects to obtain the `columnLabel` in SS.

```java
private void setIndexForAggregationProjection(final Map<String, Integer> columnLabelIndexMap) {
    for (AggregationProjection each : projectionsContext.getAggregationProjections()) {
        Preconditions.checkState(columnLabelIndexMap.containsKey(each.getColumnLabel()), "Can't find index: %s, please add alias for aggregate selections", each);
        each.setIndex(columnLabelIndexMap.get(each.getColumnLabel()));
        for (AggregationProjection derived : each.getDerivedAggregationProjections()) {
            Preconditions.checkState(columnLabelIndexMap.containsKey(derived.getColumnLabel()), "Can't find index: %s", derived);
            derived.setIndex(columnLabelIndexMap.get(derived.getColumnLabel()));
        }
    }
}
```

**Comment by strongduanmu** [comment]
This exception is caused by the fact that the aggregate function must use `group by`, which is not supported by the native database.

```
ERROR 1140 (42000): In aggregated query without GROUP BY, expression #2 of SELECT list contains nonaggregated column 'testdb_0.t_order_0.order_id'; this is incompatible with sql_mode=only_full_group_by
```

If sql is rewritten as `select count(1) from t_order group by order_id order by order_id desc;` it can be executed normally. `Can't find index` exception has been fixed by #11971.
